### PR TITLE
notification about touching yubikey when we have `Timeout`

### PR DESCRIPTION
**Pull request by Hidanio**

### CODE REVIEW

File: src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt [hunk 1]
```diff
@@ -120,6 +124,14 @@ class SshConnectionService(private var myProject: Project) : Disposable {
 
         // Disconnect the current connection, if it exists.
         disconnect()
+        val sheduler = Executors.newSingleThreadScheduledExecutor()
```
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt:127]: We have no specific situations or something else to use not standard API. Firstly, it ensures compatibility and stability across different versions of the IDE, as standard Java APIs are less likely to change compared to internal IntelliJ APIs. Secondly, it reduces dependency on IntelliJ-specific classes which might be subject to modifications or deprecation, potentially affecting the plugin's longevity. I think this is unjustified

File: src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt [hunk 2]
```diff
@@ -120,6 +124,14 @@ class SshConnectionService(private var myProject: Project) : Disposable {
...
+
+        sheduler.schedule({
+            ApplicationManager.getApplication().invokeLater{
```
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt]: Fixed

File: src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt]: Fixed

File: src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt [hunk 3]
```diff
@@ -123,6 +125,14 @@ class SshConnectionService(private var myProject: Project) : Disposable {
         // Disconnect the current connection, if it exists.
         disconnect()
 
+        val scheduler = ConcurrencyUtil.newSingleScheduledThreadExecutor("YubiKeyTouchingNotification")
+        scheduler.schedule({
+            ApplicationManager.getApplication().invokeLater {
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt:130]: You can just use the `invokeLater` from `com.intellij.openapi.application.invokeLater`

Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt:130]: This is `Obsolete`

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**